tests/functional/test_snap_dcgm.py

```diff
@@ -225,3 +237,45 @@ def test_valid_metric(self) -> None:
         self.set_config(self.service, self.config, metric_file)
         self.check_metric_config(metric_file_path)
         _check_endpoint(self.endpoint)
+
+    @classmethod
+    @pytest.fixture
+    def dependency_setup(cls):
+        cls.nv_hostengine_port_config = "nv-hostengine-port"
+        cls.nv_hostengine_service = "dcgm.nv-hostengine"
+        cls.dcgm_exporter_service = "dcgm.dcgm-exporter"
+
+        old_value = cls.get_config(cls.nv_hostengine_port_config)
+
+        yield
+
+        # Revert back
+        cls.set_config(cls.nv_hostengine_service, cls.nv_hostengine_port_config, old_value)
```

Comment:
You can use the unset_config here

Reply:
I think it is a little more explicit for the case of setting the port or the bind address

Reply:
Let's keep the consistency with the other tests. You are checking later the bind_config with the old_value and that is declarative enough

Reply:
Okay, agreed